docs/NOTIFICATION_MANAGEMENT.md

```diff
@@ -0,0 +1,88 @@
+the other hand, uses can use new command `tower-cli job_template associate_notification` 
+and `tower-cli job_tempate disassociate_notification` to (dis)associate an existing 
+notification to/from a job template.
+
+### Basic Operations
+
+CRUD operations on notification templates are basically the same as that of typicall
+existing resources, e.g. `inventory`. There are, however, some points due to the nature
+of notification templates that needs to be careful with:
+
+* There are two options in `create`, `--job-template` and `--status`, which controls 
+the create mode: providing these options will create a new notification template and 
+associate it with the specified job template. Notification templates will be created
+isolatedly if these options are not provided. You can find more information in Tower's
+official documentation.
+* When looking at the help text of certain notification template commands, note a large
+portion of the available options are prefixed by a label like `[slack]`. These are special
+configuration-related options which are composing elements of a notification template's
+destination configuration. You can find more information about those options in Tower's 
+official documentations. These options plus `--notification-configuration` option form
+configuration-related options.
+* Configuration-related options are *not* functioning options in `get`, `list` and `delete`,
+meaning they will be ignored under these commands even provided. 
+* The label prefixing configuration-related options indicate the type of notification
+destination this option is used for. When creating a new notification template with certain
+destination type (controlled by `--notification-type` option), *all non-default* related 
+configuration-related options must be provided.
+* When modifying an existing notification template, not every configuration-related
+option has to be provided. But if this modification modifies destination type, *all non-default* 
+related configuration-related options must be provided.
+* `--notification-configuration` option provides a json file specifying the configuration
+details. All other configuration-related options will be ignored if `--notification-configuration``
+is provided.
+
+### Detailed Example
+
+```bash
+# Create a notification template in isolation.
+tower-cli notification create --name foo --description bar --notification-type slack --channels a --channels b --token hey
+
+# Create a notification template under an existing job template.
+tower-cli notification create --name foo --description bar --notification-type slack --channels a --channels b --token hey --job-template 5
+
+# Get exactly one notification template.
+tower-cli notification get --name foo
+
+# Get a list of notification templates under certain criteria.
+tower-cli notification list --notification-type irc
+
+# Modify an existing notification.
+tower-cli notification modify --description foo --token hi 17
```

Comment:
Everything here is great, by the way. The only thing I notice as sticking out to me is the fact that `token` is a protected field, and if it is not given, the modification gets sent with `$encrypted$` as its value. That's a fundamental limitation of the server, and there's literally no way we can let the user modify anything without them also giving a new token.

So the only thing I could see being done about it would be to raise an error if the token is not provided in the`modify` command. This might be complicated if the encrypted fields are specific to the type. In that case, we could probably leave it as a TODO for later.